src/recoilState/merkleDistributor.ts

```diff
@@ -0,0 +1,22 @@
+import {
+  MerkleDistributorInfo,
+  parseBalanceMap,
+} from 'helpers/merkle-distributor/parse-balance-map';
+import { selectorFamily } from 'recoil';
+
+import { rReceivedGiftsByUser } from './receivedGiftsByUser';
+
+import { IRecoilGetParams } from 'types';
+
+export const rMerkleDistributor = selectorFamily<
+  MerkleDistributorInfo | undefined,
+  number
+>({
+  key: 'rReceivedGiftsByUser',
+  get: (epochId: number) => ({ get }: IRecoilGetParams) => {
+    const receivedGiftsByUser = get(rReceivedGiftsByUser(epochId));
+    if (Object.keys(receivedGiftsByUser).length !== 0) {
```

Comment:
is it possible to create an empty tree?
I want to move away from having `| undefined`

Reply:
maybe that is odd, I think we will preview the distribution before upload

Reply:
No it's not possible. If we don't supply any input, we can't generate any merkle tree.